Meson build broken after googletest changes

A recent change to the meson build setup relating to googletest seems to have broken it, starting from a fresh respository:

```
> ./meson_init.sh
> ninja -C build-out/sw/sim-verilator all
```

Fails with the following

```
ninja: Entering directory `build-out/sw/sim-verilator'
[0/148] Generating googletest with a meson_exe.py custom command.
make: *** No targets specified and no makefile found. Stop.
Building googletest...
[43/148] Generating 'sw/device/lib/e287d45@@usb_ot@sta/usbdev_regs.h'.
WARNING: usbdev contains extra key reset_primary
WARNING: usbdev contains extra key other_reset_list
[68/148] Generating 'sw/device/examples/hello_usbdev/3468fca@@hello_usbdev@exe/usbdev_regs.h'.
WARNING: usbdev contains extra key reset_primary
WARNING: usbdev contains extra key other_reset_list
[143/148] Generating spiflash_export with a custom command.
FAILED: sw/device/lib/testing/empty_file_for_googletest_dependency.cc
/home/greg/.local/bin/meson --internal exe --unpickle /home/greg/work/opentitan_area3/opentitan/build-out/sw/sim-verilator/meson-private/meson_exe_bash_daf5a905982b98dd562a53de1673b00722bc367c.dat
ninja: build stopped: subcommand failed.
```

weicai i think ran into a similar issue the other day..on our end at least,
it was missing cmake.  Could that be your issue also?

On Wed, Feb 5, 2020 at 8:13 AM Greg Chadwick <notifications@github.com>
wrote:

> A recent change to the meson build setup seems to have broken it, starting
> from a fresh respository:
>
> > ./meson_init.sh
> > ninja -C build-out/sw/sim-verilator all
>
> Fails with the following
>
> ninja: Entering directory `build-out/sw/sim-verilator'
> [0/148] Generating googletest with a meson_exe.py custom command.
> make: *** No targets specified and no makefile found. Stop.
> Building googletest...
> [43/148] Generating 'sw/device/lib/e287d45@@usb_ot@sta/usbdev_regs.h'.
> WARNING: usbdev contains extra key reset_primary
> WARNING: usbdev contains extra key other_reset_list
> [68/148] Generating 'sw/device/examples/hello_usbdev/3468fca@@hello_usbdev@exe/usbdev_regs.h'.
> WARNING: usbdev contains extra key reset_primary
> WARNING: usbdev contains extra key other_reset_list
> [143/148] Generating spiflash_export with a custom command.
> FAILED: sw/device/lib/testing/empty_file_for_googletest_dependency.cc
> /home/greg/.local/bin/meson --internal exe --unpickle /home/greg/work/opentitan_area3/opentitan/build-out/sw/sim-verilator/meson-private/meson_exe_bash_daf5a905982b98dd562a53de1673b00722bc367c.dat
> ninja: build stopped: subcommand failed.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/lowRISC/opentitan/issues/1484?email_source=notifications&email_token=AAH2RSRTA7AKW5ZMMB5IKG3RBLQSBA5CNFSM4KQN3W6KYY3PNVWWK3TUL52HS4DFUVEXG43VMWVGG33NNVSW45C7NFSM4ILIGATA>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAH2RSVZENORBNT7NDFW2ILRBLQSBANCNFSM4KQN3W6A>
> .
>

--- a/apt-requirements.txt
+++ b/apt-requirements.txt
@@ -7,7 +7,6 @@ autoconf
 bison
 build-essential
 clang-format
-cmake
 curl
 flex
 g++

--- a/doc/ug/install_instructions/index.md
+++ b/doc/ug/install_instructions/index.md
@@ -15,7 +15,6 @@ This guide makes assumes the following system setup.
 * Physical access to that machine, root permissions and a graphical environment.
 * Python 3.5.2 or newer. Python 3.6+ is recommended.
 * A C++14 capable compiler. GCC 5 or Clang 3.5 should meet this requirement.
-* CMake 2.8.8 or newer.
 * 60 GB or more of disk space.
   EDA tools like Xilinx Vivado can easily take up 40 GB each.
 * We develop and test on the following Linux distributions:

--- a/sw/device/lib/testing/meson.build
+++ b/sw/device/lib/testing/meson.build
@@ -2,54 +2,6 @@
 # Licensed under the Apache License, Version 2.0, see LICENSE for details.
 # SPDX-License-Identifier: Apache-2.0
 
-sw_lib_testing_gtest_src_dir = meson.source_root() / 'sw/vendor/google_googletest'
-sw_lib_testing_gtest_lock_file = meson.source_root() / 'sw/vendor/google_googletest.lock.hjson'
-# Build dir below is $REPO_TOP/build-out/sw/${DEVICE}/sw/device/lib/testing/google_googletest
-sw_lib_testing_gtest_build_dir = meson.current_build_dir() / 'google_googletest'
-sw_lib_testing_empty_file_for_dep = 'empty_file_for_googletest_dependency.cc'
-
-# Bash script used for building googletest. Also creates an empty file
-# which is used to establish the dependency between gtest and mock_mmio.
-build_gtest_cmd = '''
-echo "Building googletest..."
-GTEST_BUILD_DIR="@0@"
-GTEST_SRC_DIR="@1@"
-EMPTY_FILE_FOR_DEP="@2@"
-# Create a directory for building gtest
-mkdir -p $GTEST_BUILD_DIR
-# Build googletest
-cmake -B $GTEST_BUILD_DIR -S $GTEST_SRC_DIR > /dev/null
-make -C $GTEST_BUILD_DIR > /dev/null
-# Create an empty file for dependencies
-touch "@OUTDIR@/$EMPTY_FILE_FOR_DEP"
-echo "Done!"
-'''.format(sw_lib_testing_gtest_build_dir, sw_lib_testing_gtest_src_dir, sw_lib_testing_empty_file_for_dep)
-
-sw_lib_testing_build_gtest = custom_target(
-  'googletest',
-  output: sw_lib_testing_empty_file_for_dep,
-  depend_files: sw_lib_testing_gtest_lock_file,
-  command: ['bash', '-e', '-c', build_gtest_cmd],
-  console: true,
-)
-
-sw_lib_testing_gtest = declare_dependency(
-  sources: [sw_lib_testing_build_gtest],
-  link_args: [
-    '-L' + sw_lib_testing_gtest_build_dir / 'lib',
-    '-lgmock',
-    '-lgmock_main',
-    '-lgtest',
-  ],
-  dependencies: dependency('threads'),
-  compile_args: [
-    # These are necessary in order to make gtest headers correctly find other
-    # gtest headers.
-    '-I' + meson.source_root() / 'sw/vendor/google_googletest/googletest/include',
-    '-I' + meson.source_root() / 'sw/vendor/google_googletest/googlemock/include',
-  ],
-)
-
 sw_lib_testing_mock_mmio = declare_dependency(
   link_with: static_library(
     'mock_mmio',
@@ -57,7 +9,7 @@ sw_lib_testing_mock_mmio = declare_dependency(
       meson.source_root() / 'sw/device/lib/base/mmio.c',
       'mock_mmio.cc',
     ],
-    dependencies: [sw_lib_testing_gtest],
+    dependencies: [sw_vendor_gtest],
     native: true,
     c_args: ['-DMOCK_MMIO'],
     cpp_args: ['-DMOCK_MMIO'],
@@ -70,7 +22,7 @@ test('mock_mmio_test', executable(
   'mock_mmio_test',
   sources: ['mock_mmio_test.cc'],
   dependencies: [
-    sw_lib_testing_gtest,
+    sw_vendor_gtest,
     sw_lib_testing_mock_mmio,
   ],
   native: true,

